### ISSUE TITLE
Detect unnecessary $ in enclosed variables

### ISSUE THREAD

**Issue by rodjek** [rodjek/puppet-lint#675]
`${$foo}` is valid code, but the second `$` is unnecessary and should be detected and fixed. This should also be added to the style guide.

/cc #674 #655

**Comment by 2fa** [comment]
What about variables manipulations inside the string?

Simple example:

```ruby
$foo = 25
$bar = "${foo}" # 25
$baz = "${$foo}" # 25
$qux = "${$foo + 200}" # 225
$quux = "${foo + 200}" # fail
```
